docs/testing/continuous-integration.md

```diff
@@ -1,43 +1,62 @@
-## Jenkins
+# Jenkins
 
-To prevent defects from entering the system or remaining in the system, 
-the tests are executed automatically by Jenkins - our continuous integration server.
+To prevent defects from entering the system or remaining in the system, the tests are executed automatically by Jenkins. Jenkins first runs php and javascript style checks through the usage of `civilint` before running unit tests.
 
-The test system runs twice. Firstly, when a change is proposed through Github 
-(as a "pull-request" or "PR"), Jenkins tests it immediately. (This takes 5-90min.) 
+Jenkins runs two different types of jobs: a pull-request (PR) job and a matrix job. The results of the tests are published at [test.civicrm.org][jenkins-test-results].
 
-Unfortunately, the full test suite would take several hours longer, so Jenkins 
-only runs some tests at this stage. 
+## PR Test Jobs
 
-The results of the tests are published at [test.civicrm.org][jenkins-test-results]
+Jenkins runs a "PR Test" job which is triggered whenever a pull request is created or updated in Github. PR test jobs can take anywhere from 5 - 90 min to complete. This works for the following repos:
 
-If the tests have failed for something that we suspect is a random failure, we
-can ask Jenkins to run the tests again by commenting in the PR "Jenkins, test
-this please" 
+* `civicrm-core`
+* `civicrm-packages`
+* `civicrm-drupal`
+* `civicrm-backdrop`
 
-Jenkins only builds a Drupal site that is built against the branch that your
-PR is modifying.  If you need to test a patch against another CMS than you
-will want to test the patch in your own environment.
+There are seven main groups or suites  of tests:
 
-## Jenkins Whitelist 
+* Karma tests - These are javascript unit tests run by Karma and live in the tests/karma directory of the civicrm-core repo.
+* CRM Tests - These are phpunit tests generally testing the CRM/ folder of civicrm-core. The tests are found in the `tests/phpunit/CRM` directory of the civicrm-core repo.
+* api_v3 Tests - These are phpunit tests that test the funtions associate with APIv3. These can be found in `tests/phpunit/api/v3` directory of the civicrm-core repo.
+* Civi Tests - These are tests that mainly test the `Civi()::` facade. They can be found in the `tests/phpunit/civi` directoy of civicrm-core repo.
+* E2E Tests - These are tests that test the integration with the Content Management System (CMS) they usually involve testing rest or Soap interfaces. These can be found in `tests/phpunit/E2E` directoy of the civicrm-core repo
+* Upgrade Tests - This is a test where it takes a known databasse state loads it into the test db and then runs the upgrade script to try and upgrade the database to the current code version.
+* Webtests  - These are selenium webtests that are run occassionally and haven't been well maintained
```

Comment:
FWIW, there's also high-level lists of tests in https://docs.civicrm.org/dev/en/latest/testing/ and https://docs.civicrm.org/dev/en/latest/testing/phpunit/ -- wonder if there's a way we can hyperlink here so that the "list of tests" doesn't need to be documented again?

Reply:
So i think the stuff above may be able to be linked to other places. The details about what is tested on what repo jobs below should stay here as that is specific to Jenkins